pkg/chrootarchive/archive.go

```diff
@@ -83,6 +83,11 @@ func untarHandler(tarArchive io.Reader, dest string, options *archive.TarOptions
 		}
 	}
 
+	rootFd, dest, err := splitPathWithRoot(root, dest)
```

Comment:
rootFD is leaked on error below. (Maybe hide that in a per-platform struct, see elsewhere; and `defer dest.close()` immediately after this succeeds?)

Reply:
> rootFD is leaked on error 

Yes :cry: Well spotted...I try to remember this but [it's hard to context switch between Rust and Go (and C)...](https://blog.skylight.io/rust-means-never-having-to-close-a-socket/)